pkg/controller/appdefinition/deploy.go

```diff
@@ -980,7 +980,10 @@ func ToFunctions(req router.Request, appInstance *v1.AppInstance, tag name.Refer
 			}
 			result = append(result, perms...)
 		}
-		result = append(result, sa, dep, pdb.ToPodDisruptionBudget(dep))
+		result = append(result, sa, dep)
+		if dep.Spec.Replicas != nil && *dep.Spec.Replicas > 1 {
```

Comment:
nit: here and other places
```suggestion
		if z.Dereference(dep.Spec.Replicas) > 1 {
```